jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -271,16 +279,19 @@ def search(self, error_on_fail=False):
             if len(self.app._catalog_source_table) == 1 or self.max_sources == 1:
                 x_coordinates = [x_coordinates]
                 y_coordinates = [y_coordinates]
+            for idx, (row, x_coord, y_coord) in enumerate(zip(self.app._catalog_source_table, x_coordinates, y_coordinates)):  # noqa:E501
+                row_info = {
```

Comment:
the loaded table has to have a 'sky centroid' column, but it is then broken up into ra and dec, which means when its written back out it cant be read back in because it doesn't have 'sky centroid'. would it be possible to add a check when loading if the table already contains a ra and dec column rather than unpacking sky centroid? as it is, a table written out by the app can not be read back in

Reply:
Does that problem pre-date this PR? Just wondering if round-tripping is in scope here (e.g., writing it out back as sky centroid as a single column in ECSV format).

Reply:
`export_table` is ultimately called which will treat the the `TableMixin` columns as a `QTable` https://github.com/spacetelescope/jdaviz/blob/48be1548a8b00be113966fbc8c354dd6fdac1026/jdaviz/core/template_mixin.py#L4790 and write the columns as is, so unless you have a column in our table that is labeled `sky_centroid`, you wouldn't be able to read it back in after exporting. This is also true if you try exporting Gaia or SDSS.

In the case of 'from file', we already have the `sky_centroid` column, so we could add the sky_centroid column with the same name to our table, export the table, and then the exported file is able to be reloaded into Imviz.

The other catalogs options could be added in a follow up effort for either writing the `sky_centroid` column to the loaded catalog table or handle it on export.